op-node/rollup/derive/ithmus_upgrade_transactions.go

```diff
@@ -0,0 +1,198 @@
+		To:                  nil,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 500_000,
+		IsSystemTransaction: false,
+		Data:                append(operatorFeeVaultDeploymentBytecode, encodedArgs...),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, deployOperatorFeeVault)
+
+	// Deploy L1 Block Proxy upgrade transaction
+	updateL1BlockProxy, err := types.NewTx(&types.DepositTx{
+		SourceHash:          updateIsthmusL1BlockProxySource.SourceHash(),
+		From:                common.Address{},
+		To:                  &predeploys.L1BlockAddr,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 50_000,
+		IsSystemTransaction: false,
+		Data:                upgradeToCalldata(isthmusL1BlockAddress),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, updateL1BlockProxy)
+
+	// Deploy Gas Price Oracle Proxy upgrade transaction
+	updateGasPriceOracleProxy, err := types.NewTx(&types.DepositTx{
+		SourceHash:          updateIsthmusGasPriceOracleSource.SourceHash(),
+		From:                common.Address{},
+		To:                  &predeploys.GasPriceOracleAddr,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 50_000,
+		IsSystemTransaction: false,
+		Data:                upgradeToCalldata(isthmusGasPriceOracleAddress),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, updateGasPriceOracleProxy)
+
+	// Deploy Operator Fee Vault Proxy upgrade transaction
+	updateOperatorFeeVaultProxy, err := types.NewTx(&types.DepositTx{
+		SourceHash:          updateOperatorFeeVaultSource.SourceHash(),
+		From:                common.Address{},
+		To:                  &predeploys.OperatorFeeVaultAddr,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 50_000,
+		IsSystemTransaction: false,
+		Data:                upgradeToCalldata(OperatorFeeVaultAddress),
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, updateOperatorFeeVaultProxy)
+
+	// Enable Isthmus transaction
+	enableIsthmus, err := types.NewTx(&types.DepositTx{
+		SourceHash:          enableIsthmusSource.SourceHash(),
+		From:                L1InfoDepositerAddress,
+		To:                  &predeploys.GasPriceOracleAddr,
+		Mint:                big.NewInt(0),
+		Value:               big.NewInt(0),
+		Gas:                 90_000,
+		IsSystemTransaction: false,
+		Data:                enableIsthmusInput,
+	}).MarshalBinary()
+
+	if err != nil {
+		return nil, err
+	}
+
+	upgradeTxns = append(upgradeTxns, enableIsthmus)
+
+	return upgradeTxns, nil
+}
+
+func encodeFeeVaultConstructorArgs(address common.Address, minWithdrawalAmount *big.Int, withdrawalNetwork uint8) ([]byte, error) {
```

Comment:
We have a solabi util that is used in consensus critical paths, the amount of features in the geth abi one makes it scarier to run in consensus. It doesnt yet support u8, not the end of the world if this doesn't change but would be a nice to have: https://github.com/ethereum-optimism/optimism/blob/1eb6d937f06458a9250573e5c36351679c99a28c/op-service/solabi/util.go#L4

Reply:
Done